### PR TITLE
cli: add a cli to verify a zkLogin signature

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

Add a way to verify zkLogin signature based on the provider and its latest JWK. 
## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui/src/keytool.rs [hunk 1]
```diff
@@ -804,6 +821,58 @@ impl KeyToolCommand {
...
+                    &Base64::decode(&sig).map_err(|e| anyhow!("Invalid base64 sig: {:?}", e))?,
+                )? {
+                    GenericSignature::ZkLoginAuthenticator(zk) => {
+                        if tx_bytes.is_none() || provider.is_none() {
+                            return Ok(CommandOutput::ZkLoginSigVerify(format!(
+                                "Parsed zkLogin signature: {:?}",
+                                zk
+                            )));
+                        }
+                        println!("Parsed zkLogin signature: {:?}", zk);
```
Comment (stefan-mysten) [inline crates/sui/src/keytool.rs:841]: Do you need to print this information for this user, or is it more of a tracing/debug kind of thing? If it's needed, then I'd suggest to put it as a field into the `CommandOutput::ZkLoginSigVerify` type, and output a table with all the needed information.

File: crates/sui/src/keytool.rs [hunk 2]
```diff
@@ -804,6 +821,58 @@ impl KeyToolCommand {
...
+                        )?;
+                        let client = reqwest::Client::new();
+                        let parsed: ImHashMap<JwkId, JWK> = fetch_jwks(
+                            &OIDCProvider::from_str(&provider.clone().unwrap()).unwrap(),
+                            &client,
+                        )
+                        .await?
+                        .into_iter()
+                        .collect();
+                        println!("Latest JWKs for provider {:?} : {:?}", provider, parsed);
```
Comment (stefan-mysten) [inline crates/sui/src/keytool.rs:855]: SImilar as the one above.

File: crates/sui/src/keytool.rs [hunk 3]
```diff
@@ -804,6 +821,58 @@ impl KeyToolCommand {
...
+                        }
+                        println!("Parsed zkLogin signature: {:?}", zk);
+
+                        let tx_data: TransactionData = bcs::from_bytes(
+                            &Base64::decode(&tx_bytes.unwrap())
+                                .map_err(|e| anyhow!("Invalid base64 tx data: {:?}", e))?,
+                        )?;
+                        let client = reqwest::Client::new();
+                        let parsed: ImHashMap<JwkId, JWK> = fetch_jwks(
+                            &OIDCProvider::from_str(&provider.clone().unwrap()).unwrap(),
```
Comment (stefan-mysten) [inline crates/sui/src/keytool.rs:849]: Not sure how this JWK works, but consider handling the unwraps properly to avoid any panics.

Comment (joyqvq) [reply crates/sui/src/keytool.rs]: revised, the top level unwrap is handled at line 845

**Review by stefan-mysten — APPROVED** [review]

File: crates/sui/src/keytool.rs (hunk 3 above)
Comment (stefan-mysten) [reply crates/sui/src/keytool.rs]: Thanks for taking care of those unwraps!

**Pull request merged**